### PR TITLE
Get wallet address

### PR DESCRIPTION
**Pull request by ftreguer**
get-wallet-address command [#1188](https://spacelephant.tpondemand.com/entity/1188-ron-jutilise-greateruns-get-wallet-address)

### CODE REVIEW

File: test/__fixtures__/commands/get-wallet-address.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    network: "devnet",
+    node: NETWORKS.devnet.url,
+    date: WALLET_CHAINMETA.timestamp.human,
+    height: WALLET_CHAINMETA.height,
+};
+
+export const shouldExit = [
+    { description: "Should exit with code 2 if network flag is not passed", args: [commandName], exitCode: 2 },
+    {
+        description: "Should exit with code 2 if network is not known",
```
Comment (Nigui) [inline test/__fixtures__/commands/get-wallet-address.ts]: it's not a command specific test

File: test/__fixtures__/commands/get-wallet-address.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        args: [commandName, "--network", "customNetwork"],
+        exitCode: 2,
+    },
+    {
+        description: "Should exit with code 2 if output format is not allowed for that command",
+        args: [commandName, "-n", "devnet", "-f", "table"],
+        exitCode: 2,
+    },
+    {
+        description: "Should exit with code 1 if UNIK from unikid is not found",
```
Comment (Nigui) [inline test/__fixtures__/commands/get-wallet-address.ts:31]: you must mock api in tests

File: src/commands/get-wallet-address.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            address,
+            publicKey,
+            chainMeta,
+            this.api.network.name,
+            this.api.getCurrentNode(),
+        );
+    }
+}
+
+export function formatOutput(
```
Comment (Nigui) [inline src/commands/get-wallet-address.ts]: why not moving this function into a `helpers` folder like you've done for `send` transaction ?

File: src/commands/get-wallet-address.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    format: string,
+    displayChainMeta: boolean,
```
Comment (Nigui) [inline src/commands/get-wallet-address.ts]: I don't think that this boolean is useful. 
You can just check if `chainMeta` parameter is not `undefined`

**Review by Nigui — CHANGES_REQUESTED** [review]
no successful test cases ?

**Comment by ftreguer** [comment]
> no successful test cases ?

No more integration tests for CLI commands, just unit tests of some specific parts

File: test/__fixtures__/commands/get-wallet-address.ts (hunk 1 above)
Comment (ftreguer) [reply test/__fixtures__/commands/get-wallet-address.ts]: Yep, it's the same on each command tests, we should create a base command tests that handle common flags errors

File: src/commands/get-wallet-address.ts (hunk 3 above)
Comment (ftreguer) [reply src/commands/get-wallet-address.ts]: I'll do it after send command merge, to apply the same pattern

File: src/commands/get-wallet-address.ts (hunk 4 above)
Comment (ftreguer) [reply src/commands/get-wallet-address.ts]: A boolean flag can't be undefined, his default value is false.

File: test/__fixtures__/commands/get-wallet-address.ts (hunk 2 above)
Comment (Nigui) [reply test/__fixtures__/commands/get-wallet-address.ts:31]: this test can crash later when we'll have that `NOT_FOUND` unikid in chain `e1615becbd39ad96344919dffa7b972f293b0a3973b05145fd6d0a1a20cac169`

**Review by Nigui — APPROVED** [review]

**Pull request merged**